{{cookiecutter.profile_name}}/lsf_submit.py

```diff
@@ -0,0 +1,187 @@
+#!/usr/bin/env python3
+"""
+lsf-submit.py
+
+Script to wrap bsub sync command for Snakemake. Uses the following job or
+cluster parameters:
+
++ `threads`
++ `resources`
+    - `mem_mb`: Expected memory requirements in megabytes. Overrides
+      cluster.mem_mb
++ `cluster`
+    - `mem_mb`: Memory that will be requested for the cluster for the job.
+      Overriden by resources.mem_mb, if present.
+      `resources`
+    - `queue`: Which queue to run job on
+    - `logdir`: Where to log stdout/stderr from cluster command
+    - `output`: Name of stdout logfile
+    - `error`: Name of stderr logfile
+    - `jobname`: Job name (with wildcards)
+
+Author: Michael B Hall
+Adapted from: https://github.com/jaicher/snakemake-sync-bq-sub
+"""
+
+import sys
+import subprocess
+from pathlib import Path
+from snakemake.utils import read_job_properties
+from typing import List
+import re
+if not __name__.startswith("tests.src."):
+    sys.path.append(str(Path(__file__).parent.absolute()))
+    from OSLayer import OSLayer
+    from CookieCutter import CookieCutter
+else:
+    from .OSLayer import OSLayer
+    from .CookieCutter import CookieCutter
+
+
+class LSF_Submit:
+    def __init__(self, argv: List[str]):
+        self._jobscript = argv[-1]
+        self._cluster_cmd = " ".join(argv[1:-1])
+        self._job_properties = read_job_properties(self._jobscript)
+        self.random_string = OSLayer.get_random_alphanumerical_string()
+
+    @property
+    def jobscript(self):
```

Comment:
I know it's annoying but could you please add type annotations?

Reply:
Done in https://github.com/leoisl/snakemake-lsf/commit/4eb7fe3e2c026e40c9e34d1a510759aca1f040cd